### PR TITLE
Java 10 fixes

### PR DESCRIPTION
**Pull request by adaszko**
Switches to Java 10 and make integration tests pass on Drone.

### CODE REVIEW

**Comment by woky** [comment]
I think that node in RPM package will not run. I guess version of emitted bytecode could be changed in sbt to make it run on older JVM.

**Comment by adaszko** [comment]
> I think that node in RPM package will not run. I guess version of emitted bytecode could be changed in sbt to make it run on older JVM.
> 
> https://stackoverflow.com/questions/3831006/setting-target-jvm-in-sbt

Do we have some RedHat/Centos machine to test it on?  I can go with a VM if not.

**Comment by adaszko** [comment]
I've extracted building a RPM into a separate Drone step.  I've tested the RPM produced on CentOS 7.  It works fine.

File: build.sbt [hunk 1]
```diff
@@ -27,7 +27,11 @@ lazy val projectSettings = Seq(
   dependencyOverrides ++= Seq(
     "io.kamon" %% "kamon-core" % kamonVersion
   ),
-  fork := true
+  fork := true,
```
Comment (woky) [inline build.sbt]: You can as well remove the fork. It's set globally in projectSettings.

File: .drone.yml [hunk 2]
```diff
@@ -85,6 +85,17 @@ pipeline:
...
       event: [ push, tag ]
 
+  make-rpm:
+    <<: *sbtenv
+    commands:
+      - env JAVAC_VERSION=1.8 sbt clean compile it:compile doc node/rpm:packageBin
+    volumes:
+      - /var/cache/rchain-build/.sbt:/root/.sbt
+      - /var/cache/rchain-build/.ivy2:/root/.ivy2
+      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (woky) [inline .drone.yml]: The docker socket mapping belongs to `sbt node/docker:publishLocal` (above).

File: .drone.yml [hunk 3]
```diff
@@ -85,6 +85,17 @@ pipeline:
     when:
       event: [ push, tag ]
 
+  make-rpm:
```
Comment (woky) [inline .drone.yml]: <bikeshed>package-rpm for consistency?</bikeshed>

Comment (adaszko) [reply .drone.yml]: ✅

File: .drone.yml (hunk 2 above)
Comment (adaszko) [reply .drone.yml]: ✅

File: build.sbt (hunk 1 above)
Comment (adaszko) [reply build.sbt]: ✅

File: .drone.yml [hunk 4]
```diff
@@ -85,6 +85,17 @@ pipeline:
...
+    <<: *sbtenv
+    commands:
+      - env JAVAC_VERSION=1.8 sbt clean compile it:compile doc node/rpm:packageBin
```
[the remark this answers is not in the document]
Comment (adaszko) [reply .drone.yml]: Removed `it:compile`.  That was my lack of understanding.

**Pull request merged**